### PR TITLE
Minor fixes to improve runtime memory usage

### PR DESCRIPTION
**Pull request by stuartwdouglas**

### CODE REVIEW

**Comment by weld-tester** [comment]
Triggering build using a merge of 2929cb329656acb64f1c911101b28a8f7e8bb343 on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/

**Comment by weld-tester** [comment]
Build 685 is now running using a merge of 2929cb329656acb64f1c911101b28a8f7e8bb343 on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/685

**Comment by weld-tester** [comment]
Build 685 outcome was SUCCESS using a merge of 2929cb329656acb64f1c911101b28a8f7e8bb343 on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/685
Public: https://hudson.jboss.org/hudson/job/Weld-2.x-pull-player-executor-osprey/685

File: impl/src/main/java/org/jboss/weld/resolution/ResolvableBuilder.java [hunk 1]
```diff
@@ -264,10 +265,13 @@ public String toString() {
 
         @Override
         public int hashCode() {
-            int result = 17;
```
Comment (jharting) [inline impl/src/main/java/org/jboss/weld/resolution/ResolvableBuilder.java]: Did you run into a code path where hashCode is called more than once for a given resolvable? Unless I am wrong it should be called exactly once most of the time.

**Comment by weld-tester** [comment]
Build 685 outcome was SUCCESS using a merge of 2929cb329656acb64f1c911101b28a8f7e8bb343 on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/685
Public: https://hudson.jboss.org/hudson/job/Weld-2.x-pull-player-executor-osprey/685

**Comment by weld-tester** [comment]
Build 685 outcome was SUCCESS using a merge of 2929cb329656acb64f1c911101b28a8f7e8bb343 on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/685
Public: https://hudson.jboss.org/hudson/job/Weld-2.x-pull-player-executor-osprey/685

**Comment by jharting** [comment]
Merged, thanks!

**Pull request closed without merging**